File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        // time :O(n^2 ) , space : O(N) // for the hashing array 
        int n  = nums.size() ;
        vector <int> ans ; 
        map<int, int >mpp ; 
        for (int i=0 ;i<n ;i++){
            int num = nums[i] ; 
            int moreValue= target- num ; 
            // finding it's presence in the hash 
            if (mpp.find(moreValue)!= mpp.end()){
                // return the indexes of the pair 
                return{mpp[moreValue]  , i } ;
            }
            mpp[num]= i ;
        }
         return ans ; 
        
    }
};

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        n =len(nums)
        left =0 
        right =n - 1 
       
        ans =[]
        for i in range(n):
            for j in range(i+1,n):
                if( (nums[i] + nums[j])== target  ):
                    ans.append(i)
                    ans.append(j)
                    return ans
        
        if(len(ans)==0 ): ans.append(-1)
        return ans
     
        